model_specific/SM/standard_model.cpp

```diff
@@ -941,91 +941,55 @@ double Standard_model::calculate_delta_alpha_s(double alphaS) const
-
-   const double scale               = get_scale();
-   const double mw_pole             = qedqcd.displayPoleMW();
-   const double mz_pole             = qedqcd.displayPoleMZ();
-   const double mt_pole             = qedqcd.displayPoleMt();
-   const double mt_drbar            = MFu(2);
-   const double mb_drbar            = MFd(2);
-   const double mh_drbar            = Mhh;
-   const double gY                  = g1 * 0.7745966692414834;
-   const double pizztMZ             = Re(self_energy_VZ_1loop(mz_pole));
-   const double piwwt0              = Re(self_energy_VWp_1loop(0.));
-   const double self_energy_w_at_mw = Re(self_energy_VWp_1loop(mw_pole));
-
-   Weinberg_angle::Self_energy_data se_data;
-   se_data.scale    = scale;
-   se_data.mt_pole  = mt_pole;
-   se_data.mt_drbar = mt_drbar;
-   se_data.mb_drbar = mb_drbar;
-   se_data.gY       = gY;
-   se_data.g2       = g2;
-
-   double pizztMZ_corrected = pizztMZ;
-   double piwwtMW_corrected = self_energy_w_at_mw;
-   double piwwt0_corrected  = piwwt0;
-
-   if (get_thresholds() > 1 && threshold_corrections.sin_theta_w > 1) {
-      pizztMZ_corrected =
-         Weinberg_angle::replace_mtop_in_self_energy_z(pizztMZ, mz_pole,
-            se_data);
-      piwwtMW_corrected =
-         Weinberg_angle::replace_mtop_in_self_energy_w(
-            self_energy_w_at_mw, mw_pole, se_data);
-      piwwt0_corrected =
-         Weinberg_angle::replace_mtop_in_self_energy_w(piwwt0, 0.,
-            se_data);
-   }
 
    const auto get_mh_pole = [&] () {
-      double mh_pole = get_physical().Mhh;
-      if (mh_pole == 0) {
-         mh_pole = input.get(Physical_input::mh_pole);
-      }
+      double mh_pole = this->get_physical().Mhh;
       if (mh_pole == 0) {
          mh_pole = Electroweak_constants::MH;
       }
       return mh_pole;
    };
 
-   Weinberg_angle::Data data;
-   data.scale               = scale;
-   data.alpha_em_drbar      = alpha_em_drbar;
-   data.alpha_s_mz          = qedqcd.displayAlphaSInput();
-   data.fermi_contant       = qedqcd.displayFermiConstant();
-   data.self_energy_z_at_mz = pizztMZ_corrected;
-   data.self_energy_w_at_mw = piwwtMW_corrected;
-   data.self_energy_w_at_0  = piwwt0_corrected;
-   data.mw_pole             = mw_pole;
-   data.mz_pole             = mz_pole;
-   data.mt_pole             = mt_pole;
-   data.mh_drbar            = mh_drbar;
-   data.mh_pole             = get_mh_pole();
-   data.gY                  = gY;
-   data.g2                  = g2;
-   data.g3                  = g3;
-
-   Weinberg_angle weinberg;
-   weinberg.disable_susy_contributions();
-   weinberg.set_number_of_loops(threshold_corrections.sin_theta_w);
-   weinberg.set_data(data);
-
-   const int error = weinberg.calculate();
-
-   theta_w = ArcSin(weinberg.get_sin_theta());
-
-   if (error)
-      problems.flag_no_sinThetaW_convergence();
-   else
-      problems.unflag_no_sinThetaW_convergence();
+   const double mt_pole = qedqcd.displayPoleMt();
```

Comment:
It seems to me that this code snippet is mostly a copy from `calculate_pole_masses`. I think it would be good to re-use the function `calculate_MVWp_pole_fit` from there. (It should get the `mh` and `mw` as parameters, I guess.)

Reply:
Maybe solved by 2e3adace5.